src/many-modules/src/_11_ledger_tokens/extended_info/visual_logo.rs

```diff
@@ -0,0 +1,271 @@
+use minicbor::{decode, encode, Decode, Decoder, Encode, Encoder};
+use num_enum::TryFromPrimitive;
+use std::cmp::Ordering;
+use std::collections::VecDeque;
+use std::ops::Deref;
+use std::sync::Arc;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum SingleVisualTokenLogo {
+    /// A single character. This is limited to a single character for now.
+    UnicodeChar(char), // TODO: Match spec. Do not limit to a single char
+    Image {
```

Comment:
If you make an `Image` type (and this becomes `Image(Image)`) which implements Encode using the derive, you'd save quite a bit of code in this file.

Can even use the `cbor_type_decl` macro for this.

Reply:
Let's discuss this offline; it doesn't appear that straightforward.

Reply:
As discussed, we're leaving it as-is for the moment.